contracts/tokenLocking/TokenLocking.sol

```diff
@@ -22,37 +22,46 @@ import "./../../lib/dappsys/math.sol";
 import "./../colony/IMetaColony.sol";
 import "./../colonyNetwork/IColonyNetwork.sol";
 import "./../common/ERC20Extended.sol";
+import "./../common/BasicMetaTransaction.sol";
 import "./../reputationMiningCycle/IReputationMiningCycle.sol";
 import "./../tokenLocking/TokenLockingStorage.sol";
 
 
-contract TokenLocking is TokenLockingStorage, DSMath { // ignore-swc-123
+contract TokenLocking is TokenLockingStorage, DSMath, BasicMetaTransaction { // ignore-swc-123
   modifier calledByColonyOrNetwork() {
     require(
-      colonyNetwork == msg.sender || IColonyNetwork(colonyNetwork).isColony(msg.sender),
+      colonyNetwork == msgSender() || IColonyNetwork(colonyNetwork).isColony(msgSender()),
       "colony-token-locking-sender-not-colony-or-network"
     );
     _;
   }
 
   modifier tokenNotLocked(address _token, bool _force) {
     if (_force) {
-      userLocks[_token][msg.sender].lockCount = totalLockCount[_token];
+      userLocks[_token][msgSender()].lockCount = totalLockCount[_token];
     }
-    require(isTokenUnlocked(_token, msg.sender), "colony-token-locking-token-locked");
+    require(isTokenUnlocked(_token, msgSender()), "colony-token-locking-token-locked");
     _;
   }
 
   modifier notObligated(address _token, uint256 _amount) {
     require(
-      sub(userLocks[_token][msg.sender].balance, _amount) >= totalObligations[msg.sender][_token],
+      sub(userLocks[_token][msgSender()].balance, _amount) >= totalObligations[msgSender()][_token],
       "colony-token-locking-excess-obligation"
     );
     _;
   }
 
   // Public functions
 
+  function getMetatransactionNonce(address userAddress) override public view returns (uint256 nonce){
```

Comment:
Is there no way to inherit these functions from elsewhere? This repetition is a buzzkill!

Reply:
The problem is they're all in different storage slots, so they each need their own declaration.